napari/layers/image/image.py

```diff
@@ -725,105 +728,76 @@ def _raw_to_displayed(self, raw):
+            if (indices[d] < 0) or (indices[d] >= self._extent_data[1][d]):
+                return
 
-            # Slice currently viewed level
-            level = self.data_level
-            indices = np.array(self._slice_indices)
-            downsampled_indices = (
-                indices[not_disp] / self.downsample_factors[level, not_disp]
-            )
-            downsampled_indices = np.round(
-                downsampled_indices.astype(float)
-            ).astype(int)
-            downsampled_indices = np.clip(
-                downsampled_indices, 0, self.level_shapes[level, not_disp] - 1
-            )
-            indices[not_disp] = downsampled_indices
-
-            scale = np.ones(self.ndim)
-            for d in self._slice_input.displayed:
-                scale[d] = self.downsample_factors[self.data_level][d]
-            self._transforms['tile2data'].scale = scale
-
-            if self._slice_input.ndisplay == 2:
-                for d in self._slice_input.displayed:
-                    indices[d] = slice(
-                        self.corner_pixels[0, d],
-                        self.corner_pixels[1, d],
-                        1,
-                    )
-                self._transforms['tile2data'].translate = (
-                    self.corner_pixels[0] * self._transforms['tile2data'].scale
-                )
-            image = self.data[level][tuple(indices)]
-            image_indices = indices
-
-            # Slice thumbnail
-            indices = np.array(self._slice_indices)
-            downsampled_indices = (
-                indices[not_disp]
-                / self.downsample_factors[self._thumbnail_level, not_disp]
-            )
-            downsampled_indices = np.round(
-                downsampled_indices.astype(float)
-            ).astype(int)
-            downsampled_indices = np.clip(
-                downsampled_indices,
-                0,
-                self.level_shapes[self._thumbnail_level, not_disp] - 1,
-            )
-            indices[not_disp] = downsampled_indices
+        # For the old experimental async code.
+        self._empty = False
 
-            thumbnail_source = self.data[self._thumbnail_level][tuple(indices)]
-        else:
-            self._transforms['tile2data'].scale = np.ones(self.ndim)
-            image_indices = self._slice_indices
-            image = self.data[image_indices]
-
-            # For single-scale we don't request a separate thumbnail_source
-            # from the ChunkLoader because in ImageSlice.chunk_loaded we
-            # call request.thumbnail_source() and it knows to just use the
-            # image itself is there is no explicit thumbnail_source.
-            thumbnail_source = None
-
-        # Load our images, might be sync or async.
-        data = self._SliceDataClass(
-            self, image_indices, image, thumbnail_source
+        # The new slicing code makes a request from the existing state and
```

Comment:
This is where the work is being triggered. This ended up being a nice surgical way of introducing request and response usage, now it will be easy to move the `execute()` call to a separate thread to prevent blocking of the main thread.